R/getItems.R

```diff
@@ -86,6 +86,8 @@ getItems <- function(x, dim = NULL, split = FALSE, full = FALSE) { # nolint
   if (!is.null(maindim)) {
     if (!(maindim %in% 1:3)) stop("Unsupported maindim (can only be 1, 2 or 3!)")
     if (dc == 0) {
+      # make sure that number of set names aggres with number of sub-dimensions
+      x <- clean_magpie(x, what = "sets", maindim = maindim)
```

Comment:
Is this affecting performance? `getItems` is called very often, right?

Reply:
yes, it is. But this particular part should not be called too often and the cleaning should be also rather quick if nothing has to be cleaned